fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/client/BulkDataClient.java

```diff
@@ -312,11 +309,139 @@ public PollingLocationResponse status(String job) throws Exception {
+
+        // Check 3 (B), else fall through (A).
+        // Default Response is Accepted (202)
+        // The Alternative Response is 200
+
+        // Check for a server-side error
+        if (HttpStatus.SC_INTERNAL_SERVER_ERROR == status.getStatusCode()) {
+            // 3.C - ERROR Condition
+            // The Server hit an error 
+            throw BulkDataExportUtil.buildOperationException(
+                    "Deleting the job has failed; the content is not abandonded", IssueType.EXCEPTION);
+        } else if (HttpStatus.SC_NO_CONTENT != status.getStatusCode()) {
+            // The Delete is unsuccessful (SC_NO_CONTENT)
+            // 3.B - STOP Condition and now step 4 in the flow.
+            status = runStopOfTenantJob(job);
+        }
+        return status;
+    }
+
+    /*
+     * stops the job (local or remote batch processor).
+     */
+    private Response.Status runStopOfTenantJob(String job) throws Exception {
+        // 4 - Check the State of the Job (batchStatus)
+        //      - batchStatus - STARTING, STARTED, STOPPING, STOPPED, FAILED, COMPLETED, ABANDONED
+
+        // 4 - Try to stop the PUT /ibm/api/batch/jobinstances/<job>?action=stop
+        //     A - SC_CONTENT_REDIRECT (302) GO_TO Location to PUT
+        //       - Stop requests sent to the batch REST API must be sent directly to the executor where the job is running.
+        //       - Link - https://www.ibm.com/support/knowledgecenter/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/rwlp_batch_rest_api.html#rwlp_batch_rest_api__STOP_requests
+        //     B - SC_ACCEPTED (202) GO_TO ACCEPTED
+        Response.Status status = Response.Status.NO_CONTENT;
+        try {
+            // NOT_LOCAL - follow location 
+            String baseUrl =
+                    properties.get(BulkDataConfigUtil.BATCH_URL).replace("jobinstances", "jobexecutions") + "/" + job
+                            + "?action=stop";
+            WebTarget target = getWebTarget(baseUrl);
+
+            // The documentation says this is a PUT, however in practice it's actually a get.
+            // @see https://www.ibm.com/support/knowledgecenter/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/rwlp_batch_rest_api.html#rwlp_batch_rest_api__http_return_codes
+            Response r = target.request().get();
+            String responseStr = r.readEntity(String.class);
+
+            if (HttpStatus.SC_MOVED_TEMPORARILY == r.getStatus()) {
+                // It's on a different server, and must be in the cluster.
+                // If not, we're going to connect to location, and try a put.
+                String location = r.getHeaderString("location");
+                if (location != null) {
+                    // No matter what, tell people we accepted the call.
+                    target = getWebTarget(location + "?action=stop");
+                    // no assignment intentionally.
+                    target.request().get();
```

Comment:
this is really a big surprise for me, because in my test, I always have to use PUT to stop the job, using GET only returns the current status of the job...

Reply:
with action=stop?  funny PUT never worked for me.

Reply:
We did a screenshare.  Paul to double check.

Reply:
Updated in enxt commit.